common/test/acceptance/tests/lms/test_oauth2.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for OAuth2 permission delegation.
+"""
+from ...pages.lms.oauth2_confirmation import OAuth2Confirmation
+from ...pages.lms.auto_auth import AutoAuthPage
+from bok_choy.web_app_test import WebAppTest
+
+from urlparse import urlparse, parse_qsl
+
+
+class OAuth2PermissionDelegationTests(WebAppTest):
+    """
+    Tests for acceptance/denial of permission delegation requests.
+    """
+
+    def setUp(self):
+        super(OAuth2PermissionDelegationTests, self).setUp()
+        self.oauth_page = OAuth2Confirmation(self.browser)
+
+    def _auth(self):
+        """
+        Authenticate the user.
+        """
+        AutoAuthPage(self.browser).visit()
+
+    def _qs(self, url):
+        """
+        Parse url's querystring into a dict.
+        """
+        return dict(parse_qsl(urlparse(url).query))
+
+    def test_error_for_invalid_scopes(self):
+        """
+        Requests for invalid scopes throw errors.
+        """
+        self._auth()
+        self.oauth_page.scopes = ('email', 'does-not-exist')
+        assert self.oauth_page.visit().is_browser_on_page()
```

Comment:
[Bok choy's visit method](http://bok-choy.readthedocs.org/en/latest/_modules/bok_choy/page_object.html#PageObject.visit) has the assertions for being on the page built-in, so instead you should simply do `self.oauth_page.visit()`